src/Apps/Order/Routes/Shipping2/Hooks/useShippingContext.ts

```diff
@@ -0,0 +1,173 @@
+import { createContext, useContext, useMemo, useState } from "react"
+import { FormikProps } from "formik"
+import { ShippingProps, ShippingRouteStep } from "Apps/Order/Routes/Shipping2"
+import {
+  FulfillmentValues,
+  ShipValues,
+} from "Apps/Order/Routes/Shipping2/FulfillmentDetails"
+import {
+  addressWithFallbackValues,
+  getDefaultUserAddress,
+  FulfillmentType,
+} from "Apps/Order/Routes/Shipping2/Utils/shippingUtils"
+import { extractNodes } from "Utils/extractNodes"
+import {
+  ParsedOrderData,
+  useParseOrderData,
+} from "Apps/Order/Routes/Shipping2/Hooks/useParseOrderData"
+
+type FulfillmentHelpers = Pick<
+  FormikProps<FulfillmentValues>,
+  "handleSubmit" | "isValid" | "values"
+>
+export interface ShippingContextProps {
+  parsedOrderData: ParsedOrderData
+  initialValues: {
+    fulfillmentDetails: FulfillmentValues
+    shippingQuotes: {
+      selectedShippingQuoteId?: string | null
+    }
+  }
+  step: ShippingRouteStep
+  helpers: {
+    fulfillmentDetails: FulfillmentHelpers & {
+      setFulfillmentFormHelpers: (helpers: FulfillmentHelpers) => void
+    }
+  }
+}
+
+export const ShippingContext = createContext<ShippingContextProps>({} as any)
+
+export const useShippingContext = () => {
+  return useContext(ShippingContext)
+}
+
+// Get information from the order and user, parsedOrderData (shared, computed context about order)
+// initial values for forms, and the current step in the shipping route
+// values for forms
+export const useComputeShippingContext = (
+  props: ShippingProps
+): ShippingContextProps => {
+  const parsedOrderData = useParseOrderData(props)
+  const initialValues = getInitialValues(props, parsedOrderData)
+
+  const { isArtsyShipping } = parsedOrderData
+
+  const step: ShippingRouteStep = useMemo(() => {
+    if (isArtsyShipping) {
+      return "shipping_quotes"
+    }
+    return "fulfillment_details"
+  }, [isArtsyShipping])
+
+  return {
+    parsedOrderData,
+    initialValues,
+    step,
+    helpers: {
+      fulfillmentDetails: useFulfillmentDetailsHelpers(),
+    },
+  }
+}
+
+const useFulfillmentDetailsHelpers = (): ShippingContextProps["helpers"]["fulfillmentDetails"] => {
```

Comment:
Is it necessary for this to be in its own hook, or can this just be in the main context?

Reply:
good call